src/main/java/com/synopsys/arc/jenkinsci/plugins/jobrestrictions/restrictions/job/StartedByMemberOfGroupRestriction.java

```diff
@@ -0,0 +1,193 @@
+
+	private final List<GroupSelector> groupList;
+	private final boolean checkGroupsFromUpstreamProjects;
+
+	private transient Set<String> acceptedGroups = null;
+
+	@DataBoundConstructor
+	public StartedByMemberOfGroupRestriction(List<GroupSelector> groupList,
+			boolean checkGroupsFromUpstreamProjects) {
+		this.groupList = groupList;
+		this.checkGroupsFromUpstreamProjects = checkGroupsFromUpstreamProjects;
+	}
+
+	public List<GroupSelector> getGroupList() {
+		return groupList;
+	}
+
+	public boolean isCheckGroupsFromUpstreamProjects() {
+		return checkGroupsFromUpstreamProjects;
+	}
+
+	private synchronized @Nonnull
+	Set<String> getAcceptedGroups() {
+		if (acceptedGroups == null) {
+			final List<GroupSelector> selectors = getGroupList();
+			acceptedGroups = new HashSet<String>(selectors.size());
+			for (GroupSelector selector : selectors) {
+				acceptedGroups.add(selector.getSelectedGroupId()); // merge
+																	// equal
+																	// entries
+			}
+		}
+		return acceptedGroups;
+	}
+
+	private boolean acceptsUser(@CheckForNull String userId) {
+		if (userId == null) {
+			return false;
+		}
+		SecurityRealm sr = Jenkins.getInstance().getSecurityRealm();
+		UserDetails userDetails = sr.loadUserByUsername(userId);
+		for (String groupId : getAcceptedGroups()) {
+			GrantedAuthority[] authorities = userDetails.getAuthorities();
+			for (GrantedAuthority auth : authorities) {
+				String authority = auth.getAuthority();
+				if (authority.equals(groupId)) {
+					return true;
+				}
+			}
+		}
+		return false;
+	}
+
+	/** package */
+	boolean canTake(@Nonnull List<Cause> causes) {
+		boolean userIdCause = false;
+		boolean rebuildCause = false;
+		boolean upstreamCause = false;
+		boolean aUserIdWasNotAccepted =false;
+		boolean userIdCauseExists=false;;
+		for (Cause cause : causes) {
+			if (cause.getClass().equals(Cause.UserIdCause.class) && !aUserIdWasNotAccepted) {
+				userIdCauseExists=true;
+				//if several userIdCauses exists, be defensive and don't allow if one is not accepted.
+				final @CheckForNull
+				String startedBy = ((Cause.UserIdCause) cause).getUserId();
+				if(acceptsUser(startedBy) )
+				{
+					userIdCause = true;
+				}
+				else
+				{
+					aUserIdWasNotAccepted=true;
+					userIdCause=false;
+				}
+			}
+			if (checkGroupsFromUpstreamProjects
+					&& cause.getClass().equals(Cause.UpstreamCause.class)) {
+
+				final List<Cause> upstreamCauses = ((Cause.UpstreamCause) cause)
+						.getUpstreamCauses();
+
+				if (canTake(upstreamCauses)) // Recursive call to iterate
+												// through all causes
+				{
+					upstreamCause = true;
+				}
+			}
+
+		}
+
+		//userId has preceedence
+		if(userIdCauseExists)
+		{
+			return userIdCause;
+		}
+		else//If no update cause exists we should also return false...
+		{
+			return upstreamCause;
+		}
+	}
+
+	@Override
+	public boolean canTake(Queue.BuildableItem item) {
+		
+		List<Action> allActions = (List<Action>) item.getAllActions();
```

Comment:
<code>getActions()</code> is enough IMO. It's hard to imagine a cause which comes from a transient action.
I'll modify the logic to make it compatible with 1.509.3

If you have any  use-cases for transient actions, I can add their support as well